include/zephyr/zbus/zbus.h

```diff
@@ -237,10 +292,10 @@ struct zbus_observer {
 #define ZBUS_SUBSCRIBER_DEFINE(_name, _queue_size)                                                 \
 	K_MSGQ_DEFINE(_zbus_observer_queue_##_name, sizeof(const struct zbus_channel *),           \
 		      _queue_size, sizeof(const struct zbus_channel *));                           \
-	_ZBUS_STRUCT_DECLARE(zbus_observer,                                                        \
-			     _name) = {ZBUS_OBSERVER_NAME_INIT(_name) /* Name field */             \
-					       .enabled = true,                                    \
-				       .queue = &_zbus_observer_queue_##_name, .callback = NULL}
+	STRUCT_SECTION_ITERABLE(zbus_observer, _name) = {                                          \
+		ZBUS_OBSERVER_NAME_INIT(_name) /* Name field */                                    \
+			.enabled = true,                                                           \
```

Comment:
It is needed. Could you elaborate more on the "they may as well implement it on their own" part?

Reply:
Yeah I'm thinking a listener may as well just have a `if (!atomic_get(&i_am_enabled)) { return; }` if needed and that's it.

Reply:
Are you suggesting using a loose variable for that? I will need to use an extern if I have a `zbus_obs_set_enable` call in a file different from where the observer was created. Is that a good practice? And how can I retrieve the correct `i_am_enabled` from several others since we will have one per each observer? Could you help me with that?

Reply:
I'm suggesting to remove it and let the caller handle the situation of "do not process this message.

Reply:
A separate bit array with the enabled information for all the observers could be a solution. But it would rely on the initial math on the `SYS_INIT`.

Reply:
I have saved 4 bytes from the struct and. Now there is a different way of storing data on the observers' metadata.